### ISSUE TITLE
Temporal (1d) padding

### ISSUE THREAD

**Issue by dhpollack** [pytorch/pytorch#2637]
It would be nice to have 1d padding. Seems easy enough to implement, just take the 2d code and remove anything related to the height then replace "Spatial"/"2d" with "Temporal"/"1d" in the names.  Anything else one has to worry about if implementing this?

**Comment by chenzhekl** [comment]
Possibly related to #1825

**Comment by braingineer** [comment]
hacky, but can do some 1d padding w/ 

```python
import torch
import torch.nn.functional as F

def pad1d(tensor, pad, permute_dims=True):
    # tensor should be in shape (batch, time, feat)
    # pad should be in shape (left, right)
    if permute_dims:
        tensor = tensor.permute(0, 2, 1).contiguous() # get features on first dim since we are padding time
    else:
        tenosr = tensor.contiguous()
    original_size = tensor.size() # (batch, feat, time)
    final_new_size = (original_size[0], original_size[1], original_size[2] + pad[0] + pad[1])
    temp_new_size = original_size[:2] + (1,) + original_size[2:]
    assert len(temp_new_size) == 4
    tensor = tensor.view(*temp_new_size)
    pad = pad + (0, 0)
    tensor = F.pad(tensor, pad)
    tensor = tensor.view(*final_new_size)
    if permute_dims:
        tensor = tensor.permute(0, 2, 1)
    return tensor
```

can even do asymmetric padding

```python
x = torch.randn(1, 2, 1)
print(x)
print(pad1d(x, (2, 1)))
```

```
(0 ,.,.) = 
  0.7670
 -0.9891
[torch.FloatTensor of size 1x2x1]

Variable containing:
(0 ,.,.) = 
  0.0000
  0.0000
  0.7670
 -0.9891
  0.0000
[torch.FloatTensor of size 1x5x1]
```

**Comment by dhpollack** [comment]
yea, I had already implemented a similarly hacky method of doing 1d padding.  But it seemed like it should be built into pytorch.  I ended up writing a pull request that adds Nd padding (#2657), but it fails the CI test for reasons I don't really understand.

**Comment by soumith** [comment]
now an Nd padding is merged into master via #2657 and it supports 1D padding as well.
